src/Symfony/Component/Messenger/Transport/RedisExt/RedisSender.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\RedisExt;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+/**
+ * @author Alexander Schranz <alexander@sulu.io>
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ *
+ * @experimental in 4.3
+ */
+class RedisSender implements SenderInterface
+{
+    private $connection;
+    private $serializer;
+
+    public function __construct(Connection $connection, SerializerInterface $serializer)
+    {
+        $this->connection = $connection;
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope): Envelope
+    {
+        $encodedMessage = $this->serializer->encode($envelope);
+
+        $this->connection->add($encodedMessage['body'], $encodedMessage['headers']);
```

Comment:
`$encoderMessage['headers'] ?? []`

Reply:
@chalasr Thx! Fixed and tests for frameworkbundle added!